lib/semian/resource.rb

```diff
@@ -5,7 +5,7 @@ class Resource #:nodoc:
     class << Semian::Resource
       # Ensure that there can only be one resource of a given type
       def instance(*args)
-        Semian.resources[args.first] ||= new(*args)
+        Semian.resources[args.first] ||= ProtectedResource.new(args.first, new(*args), nil)
       end
     end
 
```

Comment:
This seems to mock the circuits when there's no C implementation? In this case, you'll need to mock out `in_use?` here as well.